src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java

```diff
@@ -48,7 +48,8 @@
  * Final modifier on methods of final and anonymous classes.
  * </li>
  * <li>
- * Inner {@code interface} declarations that are declared as {@code static}.
+ * Type declarations nested under {@code interface} that are declared
+ * as {@code public} or {@code static}.
```

Comment:
Annotations should be mentioned somewhere as well, since you have added this token into list of acceptable tokens

Reply:
Done